lib/compat/wordpress-6.1/rest-api.php

```diff
@@ -93,3 +93,89 @@ function gutenberg_register_has_archive_on_post_types_endpoint() {
 	);
 }
 add_action( 'rest_api_init', 'gutenberg_register_has_archive_on_post_types_endpoint' );
+
+/**
+ * Use custom REST API Controller for Navigation Posts
+ *
+ * @param array $args the post type arguments.
+ * @return array the filtered post type arguments with the new REST Controller set.
+ */
+function gutenberg_update_navigation_rest_controller( $args, $post_type ) {
+	if ( in_array( $post_type, array( 'wp_navigation' ), true ) ) {
+		// Original set in
+		// https://github.com/WordPress/wordpress-develop/blob/6cbed78c94b9d8c6a9b4c8b472b88ee0cd56528c/src/wp-includes/post.php#L528.
+		$args['rest_controller_class'] = 'Gutenberg_REST_Navigation_Controller';
+	}
+	return $args;
+}
+add_filter( 'register_post_type_args', 'gutenberg_update_navigation_rest_controller', 10, 2 );
+
+function gutenberg_transform_slug_to_post_id( $response, $handler, WP_REST_Request $request ) {
+
+	$route = rest_get_route_for_post_type_items( 'wp_navigation' );
+
+	// Ignore non-Navigation REST API requests.
+	if ( ! str_starts_with( $request->get_route(), $route ) ) {
+		return $response;
+	}
+
+	// Get the slug from the request **URL** (not the request body).
+	// PUT requests will have a param of `slug` in both the URL and (potentially)
+	// in the body. In all cases only the slug in the URL should be mapped to a
+	// postId in order that the correct post to be updated can be retrived.
+	// The `slug` within the body should be preserved "as is".
+	$slug = isset( $request->get_url_params()['slug'] ) ? $request->get_url_params()['slug'] : null;
+
+	// If no slug provided assume ID and continue as normal.
+	if ( empty( $slug ) ) {
+		return $response;
+	}
+
+	$args = array(
+		'name'                   => $slug, // query by slug
+		'post_type'              => 'wp_navigation',
+		'nopaging'               => true,
+		'posts_per_page'         => '1',
+		'update_post_term_cache' => false,
+		'no_found_rows'          => true,
+	);
+
+	// Query for the Navigation Post by slug (post_name).
+	$query = new WP_Query( $args );
+
+	if ( empty( $query->post->ID ) ) {
```

Comment:
see https://github.com/WordPress/gutenberg/pull/42809/files#r962736673

Reply:
Thanks for the catch. Fixed.